test/powershell/Language/Scripting/Dynamicparameters.Tests.ps1

```diff
@@ -0,0 +1,85 @@
+﻿Describe "Dynamic parameter support in script cmdlets." -Tags "CI" {
+    BeforeAll {
+
+        Class MyTestParameter {
+            [parameter(ParameterSetName = 'pset1', position=0, mandatory=1)]
+            [string] $name            
+        }
+        
+        function foo-bar
+        {
+          [CmdletBinding()]
+          param($path)
+
+          dynamicparam {
+            if ($PSBoundParameters["path"] -contains "abc") {
+              $attributes = [System.Management.Automation.ParameterAttribute]::New()
+              $attributes.ParameterSetName = 'pset1'
+              $attributes.Mandatory = $false
+
+              $attributeCollection = [System.Collections.ObjectModel.Collection``1[System.Attribute]]::new()
+              $attributeCollection.Add($attributes)
+            
+              $dynParam1 = [System.Management.Automation.RuntimeDefinedParameter]::new("dp1", [Int32], $attributeCollection)
+
+              $paramDictionary = [System.Management.Automation.RuntimeDefinedParameterDictionary]::new()
+              $paramDictionary.Add("dp1", $dynParam1)
+            
+              return $paramDictionary
+            }
+            elseif($PSBoundParameters["path"] -contains "class") {
+                $paramDictionary = [MyTestParameter]::new()
+                return $paramDictionary
+            }
+    
+            $paramDictionary = $null
+            return $null
+          }
+
+          begin {
+            if(($paramDictionary -ne $null) -and ($paramDictionary -is [MyTestParameter]) ) {
+                $paramDictionary.name
+            }
+            elseif ($paramDictionary -ne $null) {
+              if ($paramDictionary.dp1.Value -ne $null) {
```

Comment:
fixed

---
In reply to: [77376868](https://github.com/Powershell/powershell/pull/2118#discussion_r77376868) [](ancestors = 77376868)